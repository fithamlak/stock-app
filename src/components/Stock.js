import PropTypes from 'prop-types';

const Stock = (props) => {
  const { companyName } = props;
  return (
    <>
      <p>{companyName }</p>
    </>
  );
};

Stock.propTypes = {
  companyName: PropTypes.string.isRequired,
};

export default Stock;
