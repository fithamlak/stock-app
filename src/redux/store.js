import stocksReducer from './stocks/stocksSlice';
import { configureStore } from '@reduxjs/toolkit';
const store = configureStore({
  reducer: {
    stocks: stocksReducer,
  },
});

export default store;
