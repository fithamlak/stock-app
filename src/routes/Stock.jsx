import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import Stock from '../components/Stock';

const StockPage = () => {
  const { slug } = useParams();
  const stocksArray = useSelector((store) => store.stocks.stocksArray);
  const content = stocksArray.find((stock) => stock.symbol === slug);
  const {
    symbol,
    companyName,
    sector,
    marketCap,
    beta,
    price,
    exchangeShortName,
    volume,
    lastAnnualDividend,
  } = content;

  return (
    <>
      <Stock
        symbol={symbol}
        companyName={companyName}
        sector={sector}
        marketCap={marketCap}
        beta={beta}
        price={price}
        exchangeShortName={exchangeShortName}
        volume={volume}
        lastAnnualDividend={lastAnnualDividend}
      />
    </>
  );
};

export default StockPage;
