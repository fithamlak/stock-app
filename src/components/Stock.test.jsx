import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Stock from './Stock';
import store from '../redux/store';

const SampleStock = {
  symbol: 'CRM',
  companyName: 'Salesforce, Inc.',
  marketCap: 188680000000,
  sector: 'Technology',
  beta: 1.2094,
  price: 188.68,
  exchangeShortName: 'NYSE',
  volume: 4379287,
  lastAnnualDividend: 0,
};

describe('test suite for Stock component', () => {
  it('expects to render a snapshot of the Stock component', () => {
    const stockComponent = render(
      <Provider store={store}>
        <Stock
          symbol={SampleStock.symbol}
          companyName={SampleStock.companyName}
          sector={SampleStock.sector}
          marketCap={SampleStock.marketCap}
          beta={SampleStock.beta}
          price={SampleStock.price}
          exchangeShortName={SampleStock.exchangeShortName}
          volume={SampleStock.volume}
          lastAnnualDividend={SampleStock.lastAnnualDividend}
        />
      </Provider>,
    );
    expect(stockComponent).toMatchSnapshot();
  });
});
