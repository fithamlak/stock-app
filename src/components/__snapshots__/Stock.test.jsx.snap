// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suite for Stock component expects to render a snapshot of the Stock component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3>
        Name: Salesforce, Inc.
      </h3>
      <ul>
        <li>
          Symbol: CRM
        </li>
        <li>
          Market cap: 188680000000
        </li>
        <li>
          Sector: Technology
        </li>
        <li>
          Beta: 1.2094
        </li>
        <li>
          Price: 188.68
        </li>
        <li>
          Exchange: undefined
        </li>
        <li>
          Volume: 4379287
        </li>
        <li>
          Last annual dividend: 0
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <h3>
      Name: Salesforce, Inc.
    </h3>
    <ul>
      <li>
        Symbol: CRM
      </li>
      <li>
        Market cap: 188680000000
      </li>
      <li>
        Sector: Technology
      </li>
      <li>
        Beta: 1.2094
      </li>
      <li>
        Price: 188.68
      </li>
      <li>
        Exchange: undefined
      </li>
      <li>
        Volume: 4379287
      </li>
      <li>
        Last annual dividend: 0
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
